src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IProjectFile.cs

```diff
@@ -41,6 +45,6 @@ public interface IProjectFile
 
         void SetTFM(TargetFrameworkMoniker targetTFM);
 
-        IEnumerable<string> Imports { get; }
+        ICollection<string> Imports { get; }
```

Comment:
This should stay IEnumerable. Otherwise, there appear to be two ways to add Imports (with the AddImports method or adding to the collection). We should limit it to the one recommended way (following the pattern of references would mean using Add* methods).

Reply:
Removed the Add/Remove methods and added a custom collection to support the add/remove operation